sway/commands/bind.c

```diff
@@ -196,12 +196,122 @@ static struct cmd_results *identify_key(const char* name, bool first_key,
 	return NULL;
 }
 
+static struct cmd_results *switch_binding_add_to_list(struct sway_switch_binding *binding,
+		const char *bindtype, const char *switchcombo, bool warn) {
+
+	list_t *mode_bindings = config->current_mode->switch_bindings;
+
+	// overwrite the binding if it already exists
+	bool overwritten = false;
+	for (int i = 0; i < mode_bindings->length; ++i) {
+		struct sway_switch_binding *config_binding = mode_bindings->items[i];
+		if (binding_switch_compare(binding, config_binding)) {
+			sway_log(SWAY_INFO, "Overwriting binding '%s' to `%s` from `%s`",
+					switchcombo, binding->command, config_binding->command);
+			if (warn) {
+				config_add_swaynag_warning("Overwriting binding"
+						"'%s' to `%s` from `%s`",
+						switchcombo, binding->command,
+						config_binding->command);
+			}
+			free_switch_binding(config_binding);
+			mode_bindings->items[i] = binding;
+			overwritten = true;
+		}
+	}
+
+	if (!overwritten) {
+		list_add(mode_bindings, binding);
+	}
+
+	sway_log(SWAY_DEBUG, "%s - Bound %s to command `%s`",
+			bindtype, switchcombo, binding->command);
+	return cmd_results_new(CMD_SUCCESS, NULL);
+}
+
+static struct cmd_results *switch_binding_remove_from_list(struct sway_switch_binding *binding,
+		const char *bindtype, const char *switchcombo) {
+	list_t *mode_bindings = config->current_mode->switch_bindings;
+
+	for (int i = 0; i < mode_bindings->length; ++i) {
+		struct sway_switch_binding *config_binding = mode_bindings->items[i];
+		if (binding_switch_compare(binding, config_binding)) {
+			free_switch_binding(config_binding);
+			free_switch_binding(binding);
+			list_del(mode_bindings, i);
+			sway_log(SWAY_DEBUG, "%s - Unbound %s switch",
+					bindtype, switchcombo);
+			return cmd_results_new(CMD_SUCCESS, NULL);
+		}
+	}
+
+	free_switch_binding(binding);
+	return cmd_results_new(CMD_FAILURE, "Could not find switch binding `%s`", switchcombo);
+}
+
+static struct cmd_results *binding_add_to_list(struct sway_binding *binding, list_t *mode_bindings,
+		const char *bindtype, const char *keycombo, bool warn) {
+	// overwrite the binding if it already exists
+	bool overwritten = false;
+	for (int i = 0; i < mode_bindings->length; ++i) {
+		struct sway_binding *config_binding = mode_bindings->items[i];
+		if (binding_key_compare(binding, config_binding)) {
+			sway_log(SWAY_INFO, "Overwriting binding '%s' for device '%s' "
+					"to `%s` from `%s`", keycombo, binding->input,
+					binding->command, config_binding->command);
+			if (warn) {
+				config_add_swaynag_warning("Overwriting binding"
+						"'%s' for device '%s' to `%s` from `%s`",
+						keycombo, binding->input, binding->command,
+						config_binding->command);
+			}
+			free_sway_binding(config_binding);
+			mode_bindings->items[i] = binding;
+			overwritten = true;
+		}
+	}
+
+	if (!overwritten) {
+		list_add(mode_bindings, binding);
+	}
+
+	sway_log(SWAY_DEBUG, "%s - Bound %s to command `%s` for device '%s'",
```

Comment:
I think this should be in the `!overwritten` body. The `Overwriting binding` log message should be sufficient.